caPutLogApp/caPutJsonLogTask.cpp

```diff
@@ -0,0 +1,711 @@
+/* File:     caPutJsonLogTask.cpp
+ * Author:   Matic Pogacnik
+ * Created:  21.07.2020
+ *
+ * Implementation of a CaPutJsonLogTask class.
+ * For more information refer to the header file.
+ *
+ * Modification log:
+ * ----------------
+ * v 1.0
+ * - Initial version.
+*/
+
+// Standard library imports
+#include <cstdlib>
+#include <cstring>
+#include <string>
+
+// Epics Base imports
+#include <epicsStdio.h>
+#include <envDefs.h>
+#include <epicsString.h>
+#include <asLib.h>
+#include <logClient.h>
+#include <freeList.h>
+#include <epicsThread.h>
+#include <dbAccessDefs.h>
+#include <dbChannel.h>
+#include <epicsExit.h>
+// #include <epicsAssert.h>
+#include <yajl_gen.h>
+
+// This module imports
+#include "caPutLogAs.h"
+#include "caPutLogTask.h"
+#include "caPutJsonLogTask.h"
+
+
+#define isDbrNumeric(type) ((type) > DBR_STRING && (type) <= DBR_ENUM)
+
+static const ENV_PARAM EPICS_CA_JSON_PUT_LOG_ADDR = {epicsStrDup("EPICS_CA_JSON_PUT_LOG_ADDR"), epicsStrDup("")};
+
+CaPutJsonLogTask * CaPutJsonLogTask::instance = NULL;
+
+
+CaPutJsonLogTask *CaPutJsonLogTask::getInstance() noexcept
+{
+    if (instance == NULL) {
+        try{
+            instance = new CaPutJsonLogTask();
+        }
+        catch (...) {
+            errlogSevPrintf(errlogMajor, "caPutJsonLog: Failed to construct CA put JSON logger\n");
+            return nullptr;
+        }
+    }
+    return instance;
+}
+
+CaPutJsonLogTask::CaPutJsonLogTask()
+    : caPutJsonLogQ(1000, sizeof(LOGDATA *)),
```

Comment:
Old implementation has this hard coded here: https://github.com/epics-modules/caPutLog/blob/57dc99a8a89381e3d5b4a4b02e9f8b5645b2a8be/caPutLogApp/caPutLogAs.c#L68

But yes, this could be useful. Added in: cc1eb3a8faf49c39441a0aa2670eb452757a298b

Reply:
This variable must be set before the `caPutJsonLogInit` call. I will add this to the documentation. Default value is still 1000.